internal/rbd/nodeserver.go

```diff
@@ -299,26 +314,12 @@ func (ns *NodeServer) NodeStageVolume(
 	}
 
 	isStaticVol := parseBoolOption(ctx, req.GetVolumeContext(), staticVol, false)
-
-	// throw error when imageFeatures parameter is missing or empty
-	// for backward compatibility, ignore error for non-static volumes from older cephcsi version
-	if imageFeatures, ok := req.GetVolumeContext()["imageFeatures"]; checkImageFeatures(imageFeatures, ok, isStaticVol) {
-		return nil, status.Error(codes.InvalidArgument, "missing required parameter imageFeatures")
-	}
-
 	rv, err := populateRbdVol(ctx, req, cr, req.GetSecrets())
+	defer rv.Destroy()
```

Comment:
This seems to be the only caller, thanks for adding the `.Destroy()`. Maybe only do this when no error occurred and do not return a (potential) incomplete `rv`, that will keep the function cleaner to use (or document the behaviour with a clear warning?).

Reply:
done. Do internal cleanup in case of error and caller will Destroy when no error is returned.